src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +395,32 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, key, val):
+    """Add target information to run_info under target_data.
+
+    Will Recursively construct a nested dict with the keys provided
+
+    :param string scope: The running scope
+    :param string target: The target that we want to store info for
+    :param list of string key: The key for the info being stored
+    :param dict or string val: The value of the info being stored
+
+    :API: public
+    """
+    def insert_key_value(keys, val_item, index):
```

Comment:
This is quite confusing. It's generally not clear what this helper function does. In particular, the first param is called "keys", and you index into it, implying that it's a list, but below you pass into it a value named "key", singular.  Copious comments explaining what the types of the various arguments are, and what this function does with them, and why, would be very helpful...